### PR TITLE
constants and config refactor

### LINKED ISSUE #890: Constants.json in common-files holds some variables that are not constant in the event of a contract update
https://github.com/EYBlockchain/nightfall_3/blob/chait/dupNullifierError/common-files/constants/constants.json

- PROPOSE_BLOCK_TYPES
- SUBMIT_TRANSACTION_TYPES
- BLOCK_TYPES (might require a contract migrate potentially but it is best to have this configureable too)
- TRANSACTION_TYPES (might require a contract migrate potentially but it is best to have this configureable too)

### PR DESCRIPTION
**Pull request by LijuJoseJJ**
this PR fixes #890

### CODE REVIEW

File: config/default.js [hunk 1]
```diff
@@ -429,4 +429,13 @@ module.exports = {
   // assumption is if LOCAL_PROPOSER is true, wallet UI app
   // is running in local machine
   isLocalRun: process.env.LOCAL_PROPOSER === 'true',
+  BLOCK_TYPES: '(uint48,address,bytes32,uint256,bytes32,bytes32)',
```
Comment (signorecello) [inline config/default.js:432]: for organizational sake, can we make these types into a nested object? like:

```
{
...,
"TYPES": {
    "BLOCK": '(uint48,address,bytes32,uint256,bytes32,bytes32)',
    etc
},
...
}
```

**Review by signorecello — CHANGES_REQUESTED** [review]

File: config/default.js (hunk 1 above)
Comment (ChaitanyaKonda) [reply config/default.js:432]: To be specific to their exact function of function/structure signatures, `SIGNATURES` would be very apt

**Review by ChaitanyaKonda — APPROVED** [review]

File: config/default.js (hunk 1 above)
Comment (LijuJoseJJ) [reply config/default.js]: done

**Review by signorecello — APPROVED** [review]
LGTM, thanks for the changes

**Review by RogerTaule — CHANGES_REQUESTED** [review]
The following constants are unused: BLOCK_PROPOSED_EVENT_NAME, ZKP_KEY_LENGTH and NODE_HASHLENGTH.

The following config variables are unused: NULLIFIER_COLLECTION, WALLETS_COLLECTION, PEERS_COLLECTION, PROOF_QUEUE and ELLIGATOR2. 

Could you please remove them? Thanks!

**Review by RogerTaule — APPROVED** [review]
LGTM

File: nightfall-optimist/src/services/block-utils.mjs [hunk 2]
```diff
@@ -5,7 +5,7 @@ import Web3 from 'web3';
 // of the block object isn't.  They can thus be called directly when instantiating the Block class
 // would be problematic because of its reliance on the Optimist database.
 
-const { BLOCK_TYPES } = constants;
+const { SIGNATURES } = constants;
```
Comment (RogerTaule) [inline nightfall-optimist/src/services/block-utils.mjs:8]: Signatures variable is stored in config and not in constants
```suggestion
const { SIGNATURES } = config;
```

**Review by RogerTaule — CHANGES_REQUESTED** [review]

File: nightfall-optimist/src/services/block-utils.mjs (hunk 2 above)
Comment (LijuJoseJJ) [reply nightfall-optimist/src/services/block-utils.mjs]: done

**Review by RogerTaule — APPROVED** [review]

**Pull request merged**